src/common/util/include/openvino/util/monitors/cpu_device.hpp

```diff
@@ -0,0 +1,30 @@
+// Copyright (C) 2019-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <deque>
+#include <memory>
+#include <vector>
+
+#include "openvino/util/monitors/idevice.hpp"
+
+namespace ov {
+namespace util {
+class CPUDevice : public IDevice {
+    // This class is used to monitor CPU performance data.
+    // It uses the PerformanceImpl class to get the actual performance data.
+    // The user only needs to call the get_utilization() method to get the performance data.
+public:
+    CPUDevice(int n_cores = 0);
+    virtual ~CPUDevice() = default;
+    std::map<std::string, float> get_utilization() override;
+
+private:
+    int n_cores;
+    class PerformanceImpl;
+    std::shared_ptr<PerformanceImpl> m_perf_impl = nullptr;
```

Comment:
The `CPUDevice` class looks like impl of some device monitor but it holds another impl, maybe it can be simplified.
see ov::Tensor implementation and internal tensor types like tensor view, ROI, allocated tensor etc.